podman_compose.py

```diff
@@ -3235,8 +3234,20 @@ async def compose_up(compose: PodmanCompose, args: argparse.Namespace) -> int |
         build_exit_code = await compose.commands["build"](compose, build_args)
         if build_exit_code != 0:
             log.error("Build command failed")
-            if not args.dry_run:
-                return build_exit_code
+            return build_exit_code
```

Comment:
Rather large commit just to fix pylint warnings

Reply:
I got the following warning from pylint:

```
podman_compose.py:3215:12: W1203: Use lazy % formatting in logging functions (logging-fstring-interpolation)
podman_compose.py:3220:12: W1203: Use lazy % formatting in logging functions (logging-fstring-interpolation)
podman_compose.py:3283:0: R0911: Too many return statements (7/6) (too-many-return-statements)
```

And since build image and pull image are both prepare images before running, I split it out to the `prepare_images`.

Any idea to fix the warning with less changes?

Reply:
Hi, p12tic,

I just split the last commit into two commits, but i don't know if this is what you want. 
Please check it.